homeassistant/components/homekit/__init__.py

```diff
@@ -3,154 +3,197 @@
 For more details about this platform, please refer to the documentation at
 https://home-assistant.io/components/homekit/
 """
-import asyncio
 import logging
-import re
+from zlib import adler32
 
 import voluptuous as vol
 
-from homeassistant.const import (
-    ATTR_SUPPORTED_FEATURES, ATTR_UNIT_OF_MEASUREMENT, CONF_PORT,
-    TEMP_CELSIUS, TEMP_FAHRENHEIT,
-    EVENT_HOMEASSISTANT_START, EVENT_HOMEASSISTANT_STOP)
 from homeassistant.components.climate import (
     SUPPORT_TARGET_TEMPERATURE_HIGH, SUPPORT_TARGET_TEMPERATURE_LOW)
+from homeassistant.const import (
+    ATTR_CODE, ATTR_SUPPORTED_FEATURES, ATTR_UNIT_OF_MEASUREMENT,
+    CONF_PORT, TEMP_CELSIUS, TEMP_FAHRENHEIT,
+    EVENT_HOMEASSISTANT_START, EVENT_HOMEASSISTANT_STOP)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entityfilter import FILTER_SCHEMA
 from homeassistant.util import get_local_ip
 from homeassistant.util.decorator import Registry
+from .const import (
+    DOMAIN, HOMEKIT_FILE, CONF_AUTO_START, CONF_ENTITY_CONFIG, CONF_FILTER,
+    DEFAULT_PORT, DEFAULT_AUTO_START, SERVICE_HOMEKIT_START)
+from .util import (
+    validate_entity_config, show_setup_message)
 
 TYPES = Registry()
 _LOGGER = logging.getLogger(__name__)
 
-_RE_VALID_PINCODE = r"^(\d{3}-\d{2}-\d{3})$"
-
-DOMAIN = 'homekit'
-REQUIREMENTS = ['HAP-python==1.1.7']
-
-BRIDGE_NAME = 'Home Assistant'
-CONF_PIN_CODE = 'pincode'
-
-HOMEKIT_FILE = '.homekit.state'
-
-
-def valid_pin(value):
-    """Validate pin code value."""
-    match = re.match(_RE_VALID_PINCODE, str(value).strip())
-    if not match:
-        raise vol.Invalid("Pin must be in the format: '123-45-678'")
-    return match.group(0)
+REQUIREMENTS = ['HAP-python==1.1.7', 'pypng==0.0.18']
```

Comment:
We should not add ta requirement on pypng just for the QR code. In that case just show the code.

Reply:
I removed the QR Code and `pypng` for now and will add it in another PR as an HTTP View later.